lepton/elf.go

```diff
@@ -0,0 +1,140 @@
+package lepton
+
+import (
+	"path/filepath"
+	"debug/elf"
+	"os"
+)
+
+func lookupFile(path string) (string, error) {
+	_, err := os.Lstat(path)
+	if err == nil || !os.IsNotExist(err) {
+		// file exists or other error
+		return path, err
+	}
+
+	// try to find in $NANOS_TARGET_ROOT
+	//targetRoot := os.Getenv("NANOS_TARGET_ROOT")
+	targetRoot := ""
+	if targetRoot == "" {
+		return path, err
+	}
+
+	var targetPath string
+	currentPath := path
+	for {
+		targetPath = filepath.Join(targetRoot, currentPath)
+		fi, err := os.Lstat(targetPath)
+		if err != nil {
+			return path, err
+		}
+		if fi.Mode() & os.ModeSymlink == 0 {
+			break;
+		}
+
+		currentPath, err = os.Readlink(targetPath)
+		if err != nil {
+			return path, err
+		}
+		if currentPath[0] != '/' {
+			// relative symlinks are ok
+			break;
+		}
+	}
+
+	// fmt.Printf("%s -> %s\n", path, targetPath)
+	return targetPath, nil
+}
+
+func findLib(libDirs []string, path string) (string, error) {
+	if path[0] == '/' {
```

Comment:
When is path[0] == '/'? Is it when on Mac?

Reply:
@tijoytom I've just tried to follow ld.so(8) man page:

```
If a shared object dependency does not contain a slash, then it is searched for in the following order:
...
```